### PR TITLE
Implement SkiaImage Save

### PR DESCRIPTION
**Pull request by lindexi**
### Description of Change

I implement the Save and SaveAsync method in SkiaImage .

The origin PR: https://github.com/dotnet/Microsoft.Maui.Graphics/pull/475

### CODE REVIEW

File: src/Graphics/src/Graphics.Skia/SkiaImage.cs [hunk 1]
```diff
@@ -100,38 +100,41 @@ public IImage Resize(float width, float height, ResizeMode resizeMode = ResizeMo
...
+			// Skia quality range from 0-100, this is supported by jpeg and webp. Higher values correspond to improved visual quality, but less compression.
+			const int MaxSKQuality = 100; 
+			var skQuality = (int)(MaxSKQuality * quality);
+			SKEncodedImageFormat skEncodedImageFormat = format switch
+			{
+				ImageFormat.Png => SKEncodedImageFormat.Png,
+				ImageFormat.Jpeg => SKEncodedImageFormat.Jpeg,
+				ImageFormat.Bmp => SKEncodedImageFormat.Bmp,
+				ImageFormat.Gif => SKEncodedImageFormat.Gif,
+				ImageFormat.Tiff => throw new PlatformNotSupportedException($"Do not support Tiff format in Skia"),
```
[the remark this answers is not in the document]
Comment (lindexi) [reply src/Graphics/src/Graphics.Skia/SkiaImage.cs]: Thank you @MartyIX

File: src/Graphics/src/Graphics.Skia/SkiaImage.cs [hunk 2]
```diff
@@ -100,38 +100,41 @@ public IImage Resize(float width, float height, ResizeMode resizeMode = ResizeMo
...
+		{
+			// Skia quality range from 0-100, this is supported by jpeg and webp. Higher values correspond to improved visual quality, but less compression.
+			const int MaxSKQuality = 100; 
+			var skQuality = (int)(MaxSKQuality * quality);
+			SKEncodedImageFormat skEncodedImageFormat = format switch
+			{
+				ImageFormat.Png => SKEncodedImageFormat.Png,
+				ImageFormat.Jpeg => SKEncodedImageFormat.Jpeg,
+				ImageFormat.Bmp => SKEncodedImageFormat.Bmp,
+				ImageFormat.Gif => SKEncodedImageFormat.Gif,
```
Comment (mattleibow) [inline src/Graphics/src/Graphics.Skia/SkiaImage.cs:130]: I am not so sure about bmp and gif... i do kniw skia has limited save options. Probably png, jpg and webp i think.

**Review by mattleibow — COMMENTED** [review]
Thanks for moving the pr here. Just a comment on the image format and also maybe dispose the objects sooner. Like the data and stream objects to reduct the amount of memory in use and GC work.

File: src/Graphics/src/Graphics.Skia/SkiaImage.cs (hunk 2 above)
Comment (lindexi) [reply src/Graphics/src/Graphics.Skia/SkiaImage.cs:130]: Thank you @mattleibow  and I will test it.

Comment (jfversluis) [reply src/Graphics/src/Graphics.Skia/SkiaImage.cs:130]: @lindexi did you get around to testing this?

Comment (lindexi) [reply src/Graphics/src/Graphics.Skia/SkiaImage.cs:130]: @jfversluis I am testing...

Comment (lindexi) [reply src/Graphics/src/Graphics.Skia/SkiaImage.cs:130]: @mattleibow and @jfversluis The skia do not support bmp and gif and the `Encode` method will return null when the `skEncodedImageFormat` is Bmp or Gif.

Comment (lindexi) [reply src/Graphics/src/Graphics.Skia/SkiaImage.cs]: @mattleibow and @jfversluis  Could you review my PR ?

**Comment by lindexi** [comment]
@mattleibow and @jfversluis Could you re-open and review my PR ?

File: src/Graphics/src/Graphics.Skia/SkiaImage.cs [hunk 3]
```diff
@@ -100,38 +100,39 @@ public IImage Resize(float width, float height, ResizeMode resizeMode = ResizeMo
...
+		private Stream GetSkStream(ImageFormat format, float quality)
+		{
+			// Skia quality range from 0-100, this is supported by jpeg and webp. Higher values correspond to improved visual quality, but less compression.
+			const int MaxSKQuality = 100; 
+			var skQuality = (int)(MaxSKQuality * quality);
+			SKEncodedImageFormat skEncodedImageFormat = format switch
+			{
+				ImageFormat.Png => SKEncodedImageFormat.Png,
+				ImageFormat.Jpeg => SKEncodedImageFormat.Jpeg,
+				ImageFormat.Bmp or ImageFormat.Gif or ImageFormat.Tiff => throw new PlatformNotSupportedException($"Skia do not support {format} format."),
```
Comment (mattleibow) [inline src/Graphics/src/Graphics.Skia/SkiaImage.cs:129]:
```suggestion
				ImageFormat.Bmp or ImageFormat.Gif or ImageFormat.Tiff => throw new PlatformNotSupportedException($"Skia does not support {format} format."),
```

**Review by mattleibow — APPROVED** [review]

File: src/Graphics/src/Graphics.Skia/SkiaImage.cs (hunk 3 above)
Comment (lindexi) [reply src/Graphics/src/Graphics.Skia/SkiaImage.cs]: Thank you @mattleibow

**Comment by mattleibow** [comment]
@lindexi is it possible to update from main? Something fishy is going on in the build logs saying that types are missing that you did not touch...

**Comment by lindexi** [comment]
@mattleibow Thank you and I update the branch

**Pull request merged**